src/radiation/moments.cpp

```diff
@@ -102,20 +111,26 @@ TaskStatus MomentCon2PrimImpl(T *rc) {
                      v(b, cF(ispec, 1), k, j, i) * isdetgam,
                      v(b, cF(ispec, 2), k, j, i) * isdetgam}};
 
-        Real xi = 0.0;
-        Real phi = pi;
-        if (STORE_GUESS) {
-          xi = v(b, iXi(ispec), k, j, i);
-          phi = 1.0001 * v(b, iPhi(ispec), k, j, i);
-        }
-        c.GetCovTilPiFromCon(E, covF, xi, phi, &conTilPi);
-        if (STORE_GUESS) {
-          v(b, iXi(ispec), k, j, i) = xi;
-          v(b, iPhi(ispec), k, j, i) = phi;
+        if (programming::is_specialization_of<CLOSURE, ClosureMOCMC>::value) {
+          SPACELOOP2(ii, jj) { conTilPi(ii, jj) = v(b, iTilPi(ispec, ii, jj), k, j, i); }
+        } else {
+          Real xi = 0.0;
+          Real phi = pi;
+          if (STORE_GUESS) {
```

Comment:
Why not always store guess?

Reply:
This is really @lroberts36's moment code -- he may have a use case

Reply:
There is no reason not to store the guess except that fields are not allocated for storage when we are not doing M1 (the only place a guess is required).